### ISSUE TITLE
Dynamically set cache_key is not being used in multi-tenant setup

### ISSUE THREAD

**Issue by vannut** [spatie/laravel-permission#1129]
I've got a multi-tenant app with a database per tenant.

To make sure the cached permissions don't get transferred among the different tenants i want the cache-keys to be tenant-dependent.

eg: `01ddj7nhva0t1f2ke0j7r6g19s_permissions` instead of `spatie.permissions.cache`

At runtime a middleware (first one in the web-group) checks the tennant existence, connects the database and changes some config() variables on the fly. Among these is the cache-key for this tenants permissions

```
config([
    'app.url'=> 'https://'.$this->tenant->fqdn,
    ...
    'permission.cache.key' => $this->tenant->id.'_permissions'
]);
```

When I look in telescope at the cache keys I don't see the specified tenant aware cache key; but the key statically set.

In a testController is have this:
```php
return [
   'cache_key' => config('permission.cache.key'),
   'can' => auth()->user()->can('manage-users')
]
```
Which returns the correct cache_key as a config parameter.
```
{
"cache_key": "01ddj...7r6g19s_permissions",
"can": "false"
}
```
As I use the package to retrieve the ability to `manage-users` the cache gets hit and in Telescope I see the hit being registered. But its registered with the key defined in the file permission.php instead of the dynamically set key.

I'm almost sure I'm missing something here. Any ideas why the newly set config variable is not being used?

**Comment by drbyte** [comment]
Are you using `hyn/multi-tenant`? If so, I wonder if you're running into the issue discussed here where the cache is already instantiated early and needs overriding with some namespace trickery?
https://github.com/tenancy/multi-tenant/issues/693

**Comment by vannut** [comment]
I'm actually not using that package. I read that issue and [this one](https://github.com/tenancy/multi-tenant/issues/309) and in the end found [this post.](https://github.com/tenancy/multi-tenant/issues/693#issuecomment-471367404)

Tl;dr; Because Telescope initiates the cache very early the change of config() vars is not picked up. Redis implementation of the cache has a `setPrefix()` method. So in the post above it gives a workaround to set the prefix of the redis store dynamically. 

So the problem lies with the 'early' init of the cache store, and not with the package :)

**Comment by ScriptMint** [comment]
Did you find any solution for this? I am facing same issue.

**Comment by drbyte** [comment]
> Did you find any solution for this? I am facing same issue.

When I was testing with stancl/tenancy I had no issues with this.

Are you seeing the problem "with Telescope"? Is it actually "real" independent of Telescope?
